core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1084,6 +1084,9 @@ private void forceFinishAsFailed(Throwable t) {
 
         private void doFinish() {
             if (finished.compareAndSet(false, true)) {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("finished replicating action [{}], request [{}]", actionName, replicaRequest);
+                }
```

Comment:
as this is trace, can you add more information about totalShards and failures (That's what I would be mostly interested in when debugging)?